### ISSUE TITLE
Passing arguments to @BigQueryType.fromQuery

### ISSUE THREAD

**Issue by rsandak** [spotify/scio#4449]
In the [documentation](https://spotify.github.io/scio/io/BigQuery.html#bigquerytype-fromquery) I can find an example of how to pass arguments to `@BigQueryType.fromQuery`:

```
// generate schema at compile time from a specific date
@BigQueryType.fromQuery("SELECT user, url FROM [my-project:logs.log_%s]", "20160101")
class Row

// generate new query strings at runtime
val newQuery = Row.query(args(0))
```

In the next example, even two arguments can be passed. 

Unfortunately, it doesn't work for me. `query` method accepts only one integer argument. 

```
  def main(cmdlineArgs: Array[String]): Unit = {
    val (sc, args) = ContextAndArgs(cmdlineArgs)

    val query = UserRow.query("2020-01-01 12:12:12")

    sc.typedBigQuery[UserRow]()
      .tap(x => print(x.name))

    val result = sc.run().waitUntilFinish()
  }


  @BigQueryType.fromQuery(
    """
      |  SELECT
      |   u.name
      |  FROM `project.data.user` u
      |  WHERE a.birth_date > '%s'
      |  """.stripMargin,
    "2024-01-01 12:12:12"
  )
  class UserRow
```

Additional information:
```
val scioVersion = "0.11.4"
val beamVersion = "2.38.0"`
scalaVersion := "2.13.3"
```

**Comment by rsandak** [comment]
@kellen what information do you need?

**Comment by Edmondton** [comment]
@rsandak  - hi I was wondering from the @BigQueryType.fromQuery example [here]

(https://spotify.github.io/scio/io/BigQuery.html), how do is the newQuery used ?

```
// generate schema at compile time from the latest date available in both my-project:log1.log_* and my-project:log2.log_*
@BigQueryType.fromQuery(
  "SELECT user, url, action FROM [my-project:log1.log_%s] JOIN [my-project:log2.log_%s] USING user",
  "$LATEST", "$LATEST")
class Row

// generate new query strings at runtime
val newQuery = Row.query.format(args(0), args(0))
```

**Comment by RustedBones** [comment]
Sorry this has never been answered. Hope you have resolved in the meantime.
From what I see, the query looks fine and should not have blocked the compilation. If you still experience the issue, please re-open and provide the compilation error so we can identify the root cause.

@Edmondton the `newQuery` is the query string with replaced placeholders
If `args(0)` is `one` and `args(2)` is `two`, `newQuery` is
```sql
SELECT user, url, action FROM [my-project:log1.log_one] JOIN [my-project:log2.log_two] USING user
```
basically `$LATEST` is used during compilation to generate the scala class based on the latest table partition.
When running the job, you'll need to pass the desired partition from the arguments like done above
